registrations/tasks.py

```diff
@@ -652,3 +653,50 @@ def run(self, user_id, **kwargs):
             raise
 
 pull_third_party_registrations = PullThirdPartyRegistrations()
+
+
+class SendPublicRegistrationNotifications(Task):
+    """
+    Send out a notification SMS to the CORP for public registrations not
+    converted to full registrations
+    """
+
+    def run(self):
+
+        messagesets = utils.search_messagesets(
+            {'short_name__contains': 'public'})
+        messageset_ids = [str(messageset['id']) for messageset in messagesets]
+
+        subscriptions = utils.search_subscriptions(
+            {'completed': True, 'metadata_not_has_key': 'public_notification',
+             'messageset__in': ','.join(messageset_ids)})
```

Comment:
Same here with not being able to filter the subscriptions by `messageset__in` or `metadata_not_has_key`

Reply:
I added the `metadata_not_has_key` here https://github.com/praekelt/seed-stage-based-messaging/pull/113 I'll just double check the others, thanks for the review.